### DISCUSSION TITLE
x509 certificate authentication with stompjs

### DISCUSSION THREAD

**Discussion by h0jeZvgoxFepBQ2C** [rabbitmq/rabbitmq-server#6190] [Questions]
Hi,

is there any way to authenticate via x509 certificates via stompjs? https://stomp-js.github.io/
Or do I really have to create 3000 users in my rabbitmq server, if I want to have seperate passwords/user accounts?

**Comment by michaelklishin** [comment]
STOMP (and MQTT) do [support x.509 authentication](https://github.com/rabbitmq/rabbitmq-server/blob/main/deps/rabbitmq_stomp/priv/schema/rabbitmq_stomp.schema#L166), although this is a very rarely used combination in case of STOMP.

Whether you can explicitly configure client certificates on the STOMP JS side, I'm not sure.

With modern RabbitMQ versions, [JWTs is the best alternative for Web-based clients](https://www.rabbitmq.com/oauth2.html).

[LDAP](https://www.rabbitmq.com/ldap.html) and [a custom access control service](https://github.com/rabbitmq/rabbitmq-server/tree/main/deps/rabbitmq_auth_backend_http) allow you to avoid creating any users in the RabbitMQ's internal data store.

**Comment by lukebakken** [comment]
> is there any way to authenticate via x509 certificates via stompjs?

Yes. I recently put together a client certificate example for this discussion -

https://groups.google.com/g/rabbitmq-users/c/CzgiqOUso2k/m/Xin_JooOAQAJ

https://github.com/lukebakken/rabbitmq-users-CzgiqOUso2k

You can use that as an example. If you have issues with StompJS, follow up here and we can assist you. Note that the best way for you to show us how you're progressing is to put together a repository like mine that brings up a RabbitMQ and StompJS application together.
